app/services/lawhub/authentication.rb

```diff
@@ -0,0 +1,30 @@
+class Lawhub::Authentication
+  attr_reader :credentials, :url, :body, :headers
+
+  def initialize
+    @credentials = Lawhub.credentials
+    @url = "https://login.microsoftonline.com/#{Lawhub.config.tenant_id}/oauth2/v2.0/token"
+    @body = {
+      grant_type: "client_credentials",
+      scope: Lawhub.scope
+    }
+    @headers = {
+      "Authorization" => "Basic #{credentials}",
+      "Content-Type" => "application/x-www-form-urlencoded;charset=UTF-8"
+    }
+  end
+
+  def self.access_token
+    self.new.access_token
+  end
+
+  def access_token
+    Rails.cache.fetch({lawhub: :access_token}, expires_in: 59.minutes) { request["access_token"] }
```

Comment:
Will request the token only every 59 minutes (it expires in 1 hour by default). You'll need to run this command in your terminal tho, since caching is defaulted to off in development mode:
![image](https://user-images.githubusercontent.com/1820441/107098200-48f9f300-67d4-11eb-9c37-edb9100a7bd8.png)